Hookies/Contexts/GamePlay/GameScene.swift

```diff
@@ -124,6 +127,88 @@ class GameScene: SKScene, SKPhysicsContactDelegate {
         grapplingHookButton = GrapplingHookButton(in: sceneFrame)
     }
 
+    // MARK: - Initialise Game Engine
+
+    private func initialiseGameEngine() {
+        guard let gameplayId = gameplayId else {
+            return
+        }
+
+        guard let cannonNode = self.childNode(withName: "//cannon") as? SKSpriteNode,
+            let finishingLineNode = self.childNode(withName: "//ending_line") as? SKSpriteNode
+            else {
+            return
+        }
+
+        var boltsNode = [SKSpriteNode]()
+        let bolts = getGameObject(of: GameObjectType.bolt)
+        let boltsMovable = getGameObject(of: GameObjectType.boltMovable)
```

Comment:
Instead of searching for both `bolt` and `bolt_movable`, you can use this search term to search inside the tree `self[//bolt*]` . Please refer to this article https://developer.apple.com/documentation/spritekit/sknode/searching_the_node_tree

Reply:
Doing this would affect how you write your `getGameObject` method. Maybe can add new `GameObjectType` called `GameObjectType.allBolts`